### DISCUSSION TITLE
Set password for a post

### DISCUSSION THREAD

**Discussion by Nov4ou** [cotes2020/jekyll-theme-chirpy#1688] [Q&A]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

I wnat to set password to a few of my posts, how can I do that?

### Operations you have already tried

I had search a few and try to install some extensions and failed.

**Comment by kungfux** [comment]
You're dealing with static content with no server-side application to process the authentication request. This means that any code you add can be easily manipulated and will never be secure for this purpose.

I'm not aware of any plugins. I would think the solution would be to encrypt the page content and decrypt it by JS script using the password passed in the URL parameter. But that leads to the problem that people will see garbage if they don't specify the password. Or maybe it will be possible to detect that the password is wrong and display something like "This post is password protected" instead.

However, the whole idea does not sound perfect. I would prefer to hide such posts or combine these two approaches. **It is still not going to be safe though** and require you to implement or integrate some solution like [this one](https://github.com/MaxLaumeister/pagecrypt).

**Comment by kungfux** [comment]
I was surprised to discover that there is a blog engine theme that has such a feature. I wouldn't expect a similar feature here, but if you really need it, you can check the details of [their implementation](https://theme-hope.vuejs.press/guide/feature/encrypt.html).

**Reply by huanyushi** [reply]
What about [this](https://mhexo.github.io/)? It is a Hexo theme that has such a feature. [Repo](https://github.com/D0n9X1n/hexo-blog-encrypt).

**Reply by cotes2020** [reply]
I think unless you can maintain a backend to filter user requested pages, all the pages of your static site are already exposed to the internet and setting up a password would be futile.
